src/plotman/manager.py

```diff
@@ -50,44 +50,36 @@ def phases_permit_new_job(phases, d, sched_cfg, dir_cfg):
     if len(phases) == 0:
         return True
 
-    milestone = ( sched_cfg['tmpdir_stagger_phase_major'],
-                  sched_cfg['tmpdir_stagger_phase_minor'] )
-    # 1 is the default value.
-    max_allowed_before_milestone = 1
-    if 'tmpdir_stagger_phase_limit' in sched_cfg:
-        max_allowed_before_milestone = sched_cfg['tmpdir_stagger_phase_limit']
-    if len([p for p in phases if p < milestone]) >= max_allowed_before_milestone:
+    milestone = (sched_cfg.tmpdir_stagger_phase_major, sched_cfg.tmpdir_stagger_phase_minor)
+    # tmpdir_stagger_phase_limit default is 1, as declared in configuration.py
+    if len([p for p in phases if p < milestone]) >= sched_cfg.tmpdir_stagger_phase_limit:
         return False
 
     # Limit the total number of jobs per tmp dir. Default to the overall max
     # jobs configuration, but restrict to any configured overrides.
-    max_plots = sched_cfg['tmpdir_max_jobs']
-    tmp_overrides = dir_cfg.get('tmp_overrides', None)
-    if tmp_overrides is not None and d in tmp_overrides:
-        overrides = tmp_overrides[d]
-        if 'tmpdir_max_jobs' in overrides:
-            max_plots = overrides['tmpdir_max_jobs']
+    max_plots = sched_cfg.tmpdir_max_jobs
+    if dir_cfg.tmp_overrides is not None and d in dir_cfg.tmp_overrides:
+        curr_overrides = dir_cfg.tmp_overrides[d]
+        if "tmpdir_max_jobs" in curr_overrides:
+            max_plots = curr_overrides["tmpdir_max_jobs"]
     if len(phases) >= max_plots:
         return False
 
     return True
 
 def maybe_start_new_plot(dir_cfg, sched_cfg, plotting_cfg):
-    jobs = job.Job.get_running_jobs(dir_cfg['log'])
+    jobs = job.Job.get_running_jobs(dir_cfg.log)
 
     wait_reason = None  # If we don't start a job this iteration, this says why.
 
     youngest_job_age = min(jobs, key=job.Job.get_time_wall).get_time_wall() if jobs else MAX_AGE
-    global_stagger = int(sched_cfg['global_stagger_m'] * MIN)
+    global_stagger = int(sched_cfg.global_stagger_m * MIN)
     if (youngest_job_age < global_stagger):
-        wait_reason = 'stagger (%ds/%ds)' % (
-                youngest_job_age, global_stagger)
-    elif ('global_max_jobs' in sched_cfg and
-            len(jobs) >= sched_cfg['global_max_jobs']):
-        wait_reason = 'max jobs (%d)' % sched_cfg['global_max_jobs']
+        wait_reason = 'stagger (%ds/%ds)' % (youngest_job_age, global_stagger)
```

Comment:
No need to undo them, but there are a few places where there have been unrelated changes.  I do find it is easier to handle diffs without that.  As to formatting, I vaguely 'expect' to get black in place at some point.  Though, that's just me.  I haven't gotten any team consensus on it.

Reply:
Noted! I'm personally a fan of `black` purely because it reduces developer infighting time RE: stylistic change 😅